admin/class-admin-apple-meta-boxes.php

```diff
@@ -107,13 +107,25 @@ public function do_publish( $post_id, $post ) {
 	 * @static
 	 */
 	public static function save_post_meta( $post_id ) {
-		// Save fields from the meta box
-		if ( ! empty( $_POST['apple_news_sections'] ) ) {
-			$sections = array_map( 'sanitize_text_field', $_POST['apple_news_sections'] );
+
+		// Save section override setting.
+		$mappings = get_option( Admin_Apple_Sections::TAXONOMY_MAPPING_KEY );
+		if ( ! empty( $mappings ) && empty( $_POST['apple_news_sections_by_taxonomy'] ) ) {
+			update_post_meta( $post_id, 'apple_news_section_override', true );
+		} else {
+			delete_post_meta( $post_id, 'apple_news_section_override' );
+		}
+
+		// Determine how to save sections.
+		$sections = array();
+		if ( empty( $mappings ) || empty( $_POST['apple_news_sections_by_taxonomy'] ) ) {
+			if ( ! empty( $_POST['apple_news_sections'] ) ) {
+				$sections = $_POST['apple_news_sections'];
+			}
 		} else {
-			$sections = array();
+			$sections = Admin_Apple_Sections::get_sections_for_post( $post_id );
 		}
-		update_post_meta( $post_id, 'apple_news_sections', $sections );
+		update_post_meta( $post_id, 'apple_news_sections', array_map( 'sanitize_text_field', $sections ) );
```

Comment:
If this is `$_POST` data, should verify it's an array before sanitizing or this could throw an exception.

It's OK if it's empty because there should be the option for no mappings.